synapse/tools/cryo/cat.py

```diff
@@ -55,11 +59,36 @@ def main(argv, outp=s_output.stdout):
 
         return 0
 
-    for item in cryo.slice(path, opts.offset, opts.size, opts.timeout):
-        if opts.jsonl:
-            outp.printf(json.dumps(item[1], sort_keys=True))
+    def except_wrap(it, error_str_func):
+        ''' Wrap an iterator and adds a bit of context to the exception message '''
+        item_no = 0
+        while True:
+            item_no += 1
+            try:
+                yield next(it)
+            except StopIteration:
+                return
+            except Exception as e:
+                extra_context = error_str_func(item_no)
+                e.args = (extra_context + ': ' + str(e.args[0]), ) + e.args[1:]
+                raise
+
+    if opts.ingest:
+        if opts.msgpack:
+            fd = sys.stdin.buffer
+            item_it = except_wrap(s_msgpack.iterfd(fd), lambda x: 'Error parsing item %d' % x)
         else:
-            outp.printf(pprint.pformat(item))
+            fd = sys.stdin
+            item_it = except_wrap((json.loads(s) for s in fd), lambda x: ('Failure parsing line %d of input' % x))
+        cryo.puts(path, item_it)
+    else:
+        for item in cryo.slice(path, opts.offset, opts.size, opts.timeout):
+            if opts.jsonl:
+                outp.printf(json.dumps(item[1], sort_keys=True))
+            elif opts.msgpack:
+                sys.stdout.write(s_msgpack.en(item))
+            else:
+                outp.printf(pprint.pformat(item))
```

Comment:
can we smoke in the optional offset removal here?

Reply:
ok